### PR TITLE
[G2M] date-picker - add minDate & maxDate option

### PR DESCRIPTION
**Pull request by kc-leung**
it is a feature for the current widget studio calendar/date-picker which needs to be replaced

### CODE REVIEW

**Review by sallkall — APPROVED** [review]
Nice, will add to the design guide as well.

File: src/components/date-picker.js [hunk 1]
```diff
@@ -787,6 +796,22 @@ const DatePicker = ({
         }
       }
 
+      if (formatMinDate) {
```
Comment (geoerika) [inline src/components/date-picker.js:799]: I think it is safe to combine the two ifs in one:
```
if (formatMinDate && moment(parseDay).diff(formatMinDate) < 0) {
```

File: src/components/date-picker.js [hunk 2]
```diff
@@ -787,6 +796,22 @@ const DatePicker = ({
...
+        if (moment(parseDay).diff(formatMinDate) < 0) {
+          selectedClass = 'cursor-not-allowed'
+          selectedRangeClass = 'bg-secondary-100 text-secondary-400 cursor-not-allowed'
+          disableDay = true
+        }
+      }
+
+      if (formatMaxDate) {
```
Comment (geoerika) [inline src/components/date-picker.js:807]: Same as above.

**Comment by geoerika** [comment]
Extra observation:
- I noticed that if we click on the icon in the calendar, it doesn't trigger opening the calendar. It's a it of a guessing game that we should click on the` Start` & `End date` buttons. Should we not perhaps make the icon a button as well and set it that it automatically opens & selects the start date?
Not sure how good an idea is, I just feel ux can be improved here.
@sallkall what do you think?

**Comment by sallkall** [comment]
> Extra observation:
> 
> * I noticed that if we click on the icon in the calendar, it doesn't trigger opening the calendar. It's a it of a guessing game that we should click on the` Start` & `End date` buttons. Should we not perhaps make the icon a button as well and set it that it automatically opens & selects the start date?
>   Not sure how good an idea is, I just feel ux can be improved here.
>   @sallkall what do you think?

This sounds like a personal preference thing, but I do agree we should make the icon a trigger as well. Since it's still a part of the input area and clicking anywhere inside the input area should trigger the dropdown. @kc-leung can you fix this please?

**Review by geoerika — APPROVED** [review]

**Pull request merged**